src/seedu/addressbook/Main.java

```diff
@@ -104,13 +111,16 @@ private void recordResult(CommandResult result) {
      *
      * @param command user command
      * @return result of the command
+     * @throws StorageOperationException if there are errors writing to the storage file
      */
-    private CommandResult executeCommand(Command command)  {
+    private CommandResult executeCommand(Command command) throws StorageOperationException {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
+        } catch (StorageOperationException e) {
+            throw new StorageOperationException("Error writing to storage file: file may be read-only.");
```

Comment:
Good - would be better if it tells the file location too. 

E.g. `Error writing to the file: <file location>. The file may be read-only.`